clients/java/src/main/java/io/camunda/zeebe/client/impl/util/DataSizeUtil.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.impl.util;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class DataSizeUtil {
+
+  public static final int ONE_MB = 1024 * 1024;
+  private static final Pattern PATTERN = Pattern.compile("^([+\\-]?\\d+)([a-zA-Z]{0,2})$");
+
+  private DataSizeUtil() {}
+
+  public static int parse(CharSequence text) {
+    if (text == null) {
+      throw new IllegalArgumentException("Text must not be null");
+    }
+
+    try {
+      final Matcher matcher = PATTERN.matcher(trimAllWhitespace(text));
+      if (!matcher.matches()) {
+        throw new IllegalStateException("Does not match data size pattern");
```

Comment:
We have some [guidelines for errors](https://github.com/camunda/zeebe/wiki/Error-Guidelines) to ensure they help users figure out how to resolve the problem.

❌ Please apply the guidelines

For example, you could use something like `Expected '${text}' to be a data size, but does not match pattern '${pattern}'`

Reply:
fixed